contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -3504,6 +3504,34 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 				{
 					if (sql_dialect == SQL_DIALECT_TSQL)
 						bbf_ExecDropStmt(drop_stmt);
+					/*
+					 * Prevent dropping of restricted stored procedures from PSQL dialect
+					 */
+					else if (drop_stmt->removeType == OBJECT_PROCEDURE)
```

Comment:
Suggestion:We should handle Procedure or any object irrepective of DIALECT in single function. Adding additional conditional logic complicates things for future contributors. We dont have a single convention. Current approach doesnt seem extendible. If not possible in current effort create a ticket to track the same.

Reply:
@shah-nirmit I did try using a single function for both dialects although I'm getting error while initializing Babelfish, some `DROP PROCEDURE` is failing with `Could not find relation with OID 0`, so I believe @Deepesh125 is correct here, we will need to do some pre-processing specially in case of PSQL DIALECT.